### PR TITLE
[Improvement]: Allow non admin users save grid filters

### PR DESCRIPTION
**Pull request by SamyMP**
## Changes in this pull request  
Resolves [this comment](https://github.com/pimcore/pimcore/pull/13498#issuecomment-1401733448)

Allows a non-admin user to create grid configs with saved filters. I overlooked the fact that it was admin only when I submitted [this PR](13498). I feel like it should be ok for non-admin users to use this feature as well, what do you think ?

### CODE REVIEW

File: bundles/AdminBundle/public/js/pimcore/element/helpers/gridColumnConfig.js [hunk 1]
```diff
@@ -943,7 +943,7 @@ pimcore.element.helpers.gridColumnConfig = {
         }
 
         var fields = this.getGridConfig().columns;
-        var fieldKeys = Object.keys(fields);
+        var fieldKeys = Object.entries(fields).map(([key, value]) => value.fieldConfig?.label || key);
```
Comment (lukmzig) [inline bundles/AdminBundle/public/js/pimcore/element/helpers/gridColumnConfig.js]: Hi @SamyMP, can I ask you why is this change necessary? :)

Comment (SamyMP) [reply bundles/AdminBundle/public/js/pimcore/element/helpers/gridColumnConfig.js]: Hi, it is not :)

Somehow, two of my branches were merged, I'll fix that

Comment (SamyMP) [reply bundles/AdminBundle/public/js/pimcore/element/helpers/gridColumnConfig.js]: This is fixed, sorry about that

Comment (lukmzig) [reply bundles/AdminBundle/public/js/pimcore/element/helpers/gridColumnConfig.js]: No worries, thank you for the fix :)

**Comment by lukmzig** [comment]
Thank you @SamyMP
